python/GafferUI/ColorPlugValueWidget.py

```diff
@@ -52,6 +52,8 @@ def __init__( self, plugs, **kw ) :
 
 		GafferUI.PlugValueWidget.__init__( self, self.__column, plugs, **kw )
 
+		self.__chooserAvailable = bool( sole( plug.direction() == Gaffer.Plug.Direction.In for plug in self.getPlugs() ) )
```

Comment:
We can't do a one-time initialisation for this, because `setPlugs()` can be called later to change which plugs we're showing. So I think we need to drop the member variable and defer this logic until `setColorChooserVisible()`. This does raise the question of what the behaviour should be when you've called `setColorChooserVisible( True )` when you had an output plug, and then called `setPlugs( withAnInputPlug )`. I'd suggest that we have a `__colorChooserVisible` member which stores the argument to `setColorChooserVisible()` verbatim, and then the visibily of the widget itself is `self.__colorChooserVisible and not any( p.direction() == Out for p in self.getPlugs() )`.

Reply:
Right right, I hadn't thought of that. I think it would be nice to hide the slider button for output plugs too, since it won't be active.

I implemented the chooser visibility test as you described and the button visibility in 42db3c0c5ed841a878cca311fc6cd6d648e2e20d.

I think that handles all possible cases?